garak/langservice.py

```diff
@@ -9,80 +9,80 @@
 from garak import _config, _plugins
 
 from garak.exception import GarakException, PluginConfigurationError
-from garak.translators.base import Translator
-from garak.translators.local import NullTranslator
+from garak.langproviders.base import LangProvider
+from garak.langproviders.local import Passthru
 
-translators = {}
-native_translator = None
+langproviders = {}
+native_langprovider = None
 
 
-def _load_translator(translation_service: dict = {}) -> Translator:
-    """Load a single translator based on the configuration provided."""
-    translator_instance = None
-    translator_config = {
-        "translators": {translation_service["model_type"]: translation_service}
+def _load_langprovider(language_service: dict = {}) -> LangProvider:
+    """Load a single language provider based on the configuration provided."""
+    langprovider_instance = None
+    langprovider_config = {
+        "langproviders": {language_service["model_type"]: language_service}
     }
-    logging.debug(f"translation_service: {translation_service['language']}")
-    source_lang, target_lang = translation_service["language"].split(",")
+    logging.debug(f"langauge provision service: {language_service['language']}")
+    source_lang, target_lang = language_service["language"].split(",")
     if source_lang == target_lang:
-        return NullTranslator(translator_config)
-    model_type = translation_service["model_type"]
+        return Passthru(langprovider_config)
+    model_type = language_service["model_type"]
     try:
-        translator_instance = _plugins.load_plugin(
-            path=f"translators.{model_type}",
-            config_root=translator_config,
+        langprovider_instance = _plugins.load_plugin(
+            path=f"langproviders.{model_type}",
+            config_root=langprovider_config,
         )
     except ValueError as e:
         raise PluginConfigurationError(
-            f"Failed to load '{translation_service['language']}' translator of type '{model_type}'"
+            f"Failed to load '{language_service['language']}' langprovider of type '{model_type}'"
         ) from e
-    return translator_instance
+    return langprovider_instance
 
 
-def load_translators():
-    """Loads all translators defined in configuration and validate bi-directional support"""
-    global translators, native_translator
-    if len(translators) > 0:
+def load_langproviders():
```

Comment:
Consider just making this `load()`.

```suggestion
def load():
```

As the exposed interface for `langservice` we no need to be specific about what is loading.